framework/aster-frame/src/sync/rcu/mod.rs

```diff
@@ -70,35 +159,49 @@ pub struct RcuReclaimer<P> {
 impl<P: Send + 'static> RcuReclaimer<P> {
     pub fn delay(mut self) {
         let ptr: P = unsafe {
-            let ptr = core::mem::replace(&mut self.ptr, core::mem::uninitialized());
+            // SAFETY. It is ok to have this uninitialized value because
+            // 1) Its memory won't be acccessed;
+            // 2) It will be forgotten rather than being dropped;
+            // 3) Before it gets forgtten, the code won't return prematurely or panic.
+            #[allow(clippy::mem_replace_with_uninit, clippy::uninit_assumed_init)]
+            let ptr = core::mem::replace(
+                &mut self.ptr,
+                core::mem::MaybeUninit::uninit().assume_init(),
+            );
```

Comment:
Good job on the safety documentation! I think that it's better to use `SAFETY:` instead of `SAFETY.`.

Also, instead of replacing `ptr` with the uninitialized value, I would suggest you write:
```rust
// SAFETY: ...
let ptr = unsafe { core::ptr::read(&mut self.ptr) };
core::mem::forget(self);
```
After all, `mem::replace` is [implemented by a `ptr::read` and a `ptr::write`](https://github.com/rust-lang/rust/blob/6475796a813f2153d3bfc92c6089ed0f0c1c5b46/library/core/src/mem/mod.rs#L863-L870). There is no point to perform the uninitialized write, which introduces two additional clippy warnings.